### ISSUE TITLE
Divergent behaviour for UnsafeCast rule on *nix & Windows

### ISSUE THREAD

**Issue by 3flex** [arturbosch/detekt#926]
## Expected Behavior
detekt reports the same issues regardless of platform

## Current Behavior
When running detektCheck on detekt on Windows, the following issues are found which are not identified when running on *nix:

```
UnsafeCast - [compileFromContent] at C:\projects\detekt\detekt-api\src\test\kotlin\io\gitlab\arturbosch\detekt\api\Compiler.kt:17:52
UnsafeCast - [getYmlRuleConfig] at C:\projects\detekt\detekt-cli\src\test\kotlin\io\gitlab\arturbosch\detekt\cli\ConfigAssert.kt:55:35
UnsafeCast - [compileFromContent] at C:\projects\detekt\detekt-test\src\main\kotlin\io\gitlab\arturbosch\detekt\test\KtTestCompiler.kt:22:6
```

## Steps to Reproduce (for bugs)
See AppVeyor and Travis config and output in https://github.com/arturbosch/detekt/pull/918, or run detektCheck on detekt on both Windows and a *nix OS and compare the output.

## Your Environment
* Version used: master
* Operating System and version: Windows 10 and Ubuntu 16.04
* Link to your project: See https://github.com/arturbosch/detekt/pull/918

**Comment by arturbosch** [comment]
Actually the reported issues make sense.
I do not know why they don't get reported on *nix systems. 
Basically we just use the kotlin compiler function there.
Lets just fix the issues? 
```
as KtFile ?: throw IllegalStateException("unexpected non kotlin file returned")
```
